### ISSUE TITLE
"Error running event nil handler" during TOPIC change

### ISSUE THREAD

**Issue by mikemccracken** [jorgenschaefer/circe#222]
In response to a topic change, I see the following error in `*Messages* `:
```
Error running event nil handler circe--irc-display-event: (error Unknown keyword argument "origin") (args were (#<process myservername.com<1>> "TOPIC" "auser!auser@an.ip" "#is" "A very nice new topic."))
```

**Comment by jorgenschaefer** [comment]
Hello, and thanks for the report!

Have you recently upgraded to Circe 2.0 and changed the display of topics in your personal configuration? For example by setting `circe-format-server-topic`?

**Comment by mikemccracken** [comment]
Yes, this started with upgrading to 2.0, and I do have a custom circe-format-server-topic:
```
(setq circe-format-server-topic "*** Topic change by {origin}: {topic-diff}")
```
I don't recognize that, however - it was probably copy and paste from somewhere, as much of my .emacs is. Is that customization now broken?

**Comment by jorgenschaefer** [comment]
That's the old configuration to show topic diffs instead of the topic directly. Could you change that line to the following and see if that works?

`(setq circe-format-server-topic "*** Topic change by {nick} ({userhost}): {topic-diff}")`

**Comment by mikemccracken** [comment]
I tested that by creating a new channel and setting the topic. The first time I set the topic, I got this message: 
```
Error running event nil handler circe--irc-display-event: (wrong-type-argument char-or-string-p nil) (args were (#<process a.nice.ip.com<1>> "TOPIC" "me!me@myip" "#0xdeefbebb" "what the who"))
```

So I set global debug-on-error, and then of course after that, it worked fine. However, trying again in another new channel, I see this:
```
  circe--topic-diff-split(nil)
  circe--topic-diff(nil "this is a thing")
  circe-display-topic("me" "me@myip" "TOPIC" "#bebbdeef" "this is a thing")
  apply(circe-display-topic "me" "me@myip" "TOPIC" ("#bebbdeef" "this is a thing"))
  circe--irc-display-event(#<process a.nice.ip.com<1>> "TOPIC" "me!me@myip" "#bebbdeef" "this is a thing")
  apply(circe--irc-display-event (#<process a.nice.ip.com<1>> "TOPIC" "me!me@myip" "#bebbdeef" "this is a thing"))
  irc-handler-run(#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("irc.registered" (circe--irc-conn-registered irc-handle-nickserv--registered irc-handle-auto-join--registered) "conn.disconnected" (circe--irc-conn-disconnected irc-handle-registration--disconnected) nil (circe--irc-display-event) "conn.connected" (irc-handle-registration--connected) "001" (irc-handle-registration--rpl-welcome irc-handle-state-tracking--rpl-welcome) "CAP" (irc-handle-registration--cap) "AUTHENTICATE" (irc-handle-registration--authenticate) "900" (irc-handle-registration--logged-in) "PING" (irc-handle-ping-pong--ping) "005" (irc-handle-isupport--005) "432" (irc-handle-initial-nick-acquisition--get-initial-nick) "433" (irc-handle-initial-nick-acquisition--get-initial-nick) "437" (irc-handle-initial-nick-acquisition--get-initial-nick) "PRIVMSG" (irc-handle-ctcp--privmsg irc-handle-state-tracking--PRIVMSG irc-handle-nickserv--NOTICE) "irc.ctcp" (irc-handle-ctcp--ctcp) "NOTICE" (irc-handle-ctcp--notice irc-handle-nickserv--NOTICE) "irc.ctcpreply" (irc-handle-ctcp--ctcpreply) "irc.ctcp.VERSION" (irc-handle-ctcp--ctcp-version) "irc.ctcp.CLIENTINFO" (irc-handle-ctcp--ctcp-clientinfo) "irc.ctcp.SOURCE" (irc-handle-ctcp--ctcp-source) "irc.ctcp.PING" (irc-handle-ctcp--ctcp-ping) "irc.ctcp.TIME" (irc-handle-ctcp--ctcp-time) "JOIN" (irc-handle-state-tracking--JOIN) "PART" (irc-handle-state-tracking--PART) "KICK" (irc-handle-state-tracking--KICK) "QUIT" (irc-handle-state-tracking--QUIT) "NICK" (irc-handle-state-tracking--NICK irc-handle-nickserv--NICK) "353" (irc-handle-state-tracking--rpl-namreply) "366" (irc-handle-state-tracking--rpl-endofnames) "TOPIC" (irc-handle-state-tracking--TOPIC) "331" (irc-handle-state-tracking--rpl-notopic) "332" (irc-handle-state-tracking--rpl-topic) "396" (irc-handle-auto-join--rpl-hosthidden) "nickserv.regained" (irc-handle-auto-join--nickserv-regained) "nickserv.identified" (irc-handle-auto-join--nickserv-identified) "sasl.login" (irc-handle-auto-join--sasl-login) ...)) nil #<process a.nice.ip.com<1>> "TOPIC" "me!me@myip" "#bebbdeef" "this is a thing")
  apply(irc-handler-run #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("irc.registered" (circe--irc-conn-registered irc-handle-nickserv--registered irc-handle-auto-join--registered) "conn.disconnected" (circe--irc-conn-disconnected irc-handle-registration--disconnected) nil (circe--irc-display-event) "conn.connected" (irc-handle-registration--connected) "001" (irc-handle-registration--rpl-welcome irc-handle-state-tracking--rpl-welcome) "CAP" (irc-handle-registration--cap) "AUTHENTICATE" (irc-handle-registration--authenticate) "900" (irc-handle-registration--logged-in) "PING" (irc-handle-ping-pong--ping) "005" (irc-handle-isupport--005) "432" (irc-handle-initial-nick-acquisition--get-initial-nick) "433" (irc-handle-initial-nick-acquisition--get-initial-nick) "437" (irc-handle-initial-nick-acquisition--get-initial-nick) "PRIVMSG" (irc-handle-ctcp--privmsg irc-handle-state-tracking--PRIVMSG irc-handle-nickserv--NOTICE) "irc.ctcp" (irc-handle-ctcp--ctcp) "NOTICE" (irc-handle-ctcp--notice irc-handle-nickserv--NOTICE) "irc.ctcpreply" (irc-handle-ctcp--ctcpreply) "irc.ctcp.VERSION" (irc-handle-ctcp--ctcp-version) "irc.ctcp.CLIENTINFO" (irc-handle-ctcp--ctcp-clientinfo) "irc.ctcp.SOURCE" (irc-handle-ctcp--ctcp-source) "irc.ctcp.PING" (irc-handle-ctcp--ctcp-ping) "irc.ctcp.TIME" (irc-handle-ctcp--ctcp-time) "JOIN" (irc-handle-state-tracking--JOIN) "PART" (irc-handle-state-tracking--PART) "KICK" (irc-handle-state-tracking--KICK) "QUIT" (irc-handle-state-tracking--QUIT) "NICK" (irc-handle-state-tracking--NICK irc-handle-nickserv--NICK) "353" (irc-handle-state-tracking--rpl-namreply) "366" (irc-handle-state-tracking--rpl-endofnames) "TOPIC" (irc-handle-state-tracking--TOPIC) "331" (irc-handle-state-tracking--rpl-notopic) "332" (irc-handle-state-tracking--rpl-topic) "396" (irc-handle-auto-join--rpl-hosthidden) "nickserv.regained" (irc-handle-auto-join--nickserv-regained) "nickserv.identified" (irc-handle-auto-join--nickserv-identified) "sasl.login" (irc-handle-auto-join--sasl-login) ...)) nil #<process ip<1>> "TOPIC" ("me!me@myip" "this is a thing"))
  irc-event-emit(#<process a.nice.ip.com<1>> "TOPIC" "me!me@myip" "#bebbdeef" "this is a thing")
  apply(irc-event-emit#<process a.nice.ip.com<1>> "TOPIC" "me!me@myip" ("#bebbdeef" "this is a thing"))
  irc--handle-line(#<process a.nice.ip.com<1>> ":me!me@myip TOPIC #bebbdeef :this is a thing")
  irc--filter(#<process ec2-52-25-126-98.us-west-2.compute.amazonaws.com<1>> ":me!me@myip TOPIC #bebbdeef :this is a thing
\n")
```

**Comment by jorgenschaefer** [comment]
When you upgraded Circe, did you restart Emacs, or did you just reload circe.el?

**Comment by mikemccracken** [comment]
I've restarted a few times since upgrading.

**Comment by jorgenschaefer** [comment]
Hah! That's indeed a bug, which happens when you join a channel that has no topic set (e.g. a new channel), and then set a new topic. Apparently, no one has done that so far. :-D This should not happen in an existing channel.

Thank you for reporting this and helping debugging the problem!

**Comment by mikemccracken** [comment]
Glad to help! Thanks for circe, I use it every day. :)

**Comment by jorgenschaefer** [comment]
Fixed in a3d3e99
